### PR TITLE
ci: add `bazel Nightlies / SQLite logic tests` job for CI

### PR DESCRIPTION
**Pull request by rickystewart**
Release note: None

### CODE REVIEW

**Comment by rickystewart** [comment]
test run: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SQLiteLogicTestsBazel/3456337?buildTab=tests&status=passed&suite=github.com%2Fcockroachdb%2Fcockroach%2Fpkg%2Fsql%2Flogictest%3A+

File: WORKSPACE [hunk 1]
```diff
@@ -121,3 +121,15 @@ http_archive(
...
+
+new_git_repository(
+    name = "com_github_cockroachdb_sqllogictest",
+    build_file_content = """
+filegroup(
+    name = "testfiles",
+    srcs = glob(["test/**/*.test"]),
+    visibility = ["//visibility:public"],
+)""",
+    commit = "96138842571462ed9a697bff590828d8f6356a2f",
```
Comment (rickystewart) [inline WORKSPACE:133]: The non-Bazel build configuration is set up to always pull from `master`. We'll have to make sure that any changes that are pushed to the `master` branch of this repo are also reflected here.

**Review by rail — APPROVED** [review]
> Reviewed 6 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/70289)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rickystewart)

**Comment by rickystewart** [comment]
bors r=rail

**Pull request merged**